src/lib/sample_metadata.rs

```diff
@@ -236,6 +238,18 @@ impl SampleMetadata {
                         distance,
                     });
                 }
+
+                // Ensure that barcode distances are greater than min_delta distance
+                if distance <= min_delta {
```

Comment:
can you refactor onto this PR: https://github.com/Singular-Genomics/singular-demux/pull/54.  Then you can use the `should_reject_delta` comparison here, so that all the logic is in `should_reject_delta`?